python/pyspark/sql/dataframe.py

```diff
@@ -3867,6 +3885,28 @@ def unionByName(self, other: "DataFrame", allowMissingColumns: bool = False) ->
         |   1|   2|   3|NULL|
         |NULL|   4|   5|   6|
         +----+----+----+----+
+
+        Example 3: Union of two DataFrames with few common columns.
+        >>> df1 = spark.createDataFrame([[1, 2, 3]], ["col0", "col1", "col2"])
+        >>> df2 = spark.createDataFrame([[4, 5, 6, 7]], ["col1", "col2", "col3","col4"])
```

Comment:
the generated result is df2 = spark.createDataFrame([[4, 5, 6]], ["col1", "col2", "col3","col4"]), need a small fix here